### PR TITLE
fix(platform): Remove blind try-except for yielding error on block

### PR DESCRIPTION
**Pull request by majdyz**
### Background

https://github.com/Significant-Gravitas/AutoGPT/pull/8267 revealed the inconsistency of the way we throw an error in the block implementation:
* Yielding `error` and abort
* Yielding `error` without aborting
* Raise an error

Any error happening in the block has already been yielding an `error`. So the simplest approach to produce an error is by simply raising it. This will mark the execution as failed. Manually yielding `error` without using an exception will be interpreted as a fully handled & completed execution.

### Changes 🏗️

* Remove all try-catch yield "error" in block implementation.
* Remove if output == "error": raise error logic on the block implementation.
* Add check to ensure all block `error` pin is a string, this will be used to expose execution error to other nodes, if needed.

Only for the new autogpt platform, currently in autogpt_platform/

- Create from scratch and execute an agent with at least 3 blocks
- Import an agent from file upload, and confirm it executes correctly
- Upload agent to marketplace
- Import an agent from marketplace and confirm it executes correctly
- Edit an agent from monitor, and confirm it executes correctly

### CODE REVIEW

**Comment by majdyz** [comment]
The changes look huge, but it's simply untabbing & removing try-catch: yield "error".
Except for the one I'll mark below.

File: autogpt_platform/backend/backend/blocks/__init__.py [hunk 1]
```diff
@@ -53,15 +53,22 @@ def all_subclasses(clz):
     if block.id in AVAILABLE_BLOCKS:
         raise ValueError(f"Block ID {block.name} error: {block.id} is already in use")
 
+    input_schema = block.input_schema.model_fields
```
Comment (majdyz) [inline autogpt_platform/backend/backend/blocks/__init__.py:56]: Non-refactoring change: 
Making sure "error" output pin is an `str` type

File: autogpt_platform/backend/backend/data/block.py [hunk 2]
```diff
@@ -272,8 +272,6 @@ def execute(self, input_data: BlockInput, **kwargs) -> BlockOutput:
         for output_name, output_data in self.run(
             self.input_schema(**input_data), **kwargs
         ):
-            if "error" in output_name:
```
Comment (majdyz) [inline autogpt_platform/backend/backend/data/block.py:275]: Non-refactoring change: 
We will still allow manual yielding of `error` that won't break the dlow.

**Comment by ntindle** [comment]
Add to `docs/content/server/new_blocks.md`

**Review by ntindle — APPROVED** [review]

**Comment by majdyz** [comment]
@ntindle updated docs/content/server/new_blocks.md, can you re-review?

**Pull request merged**